### PR TITLE
Update demo for image or video

### PR DESCRIPTION
**Pull request by ZHEQIUSHUI**

### CODE REVIEW

**Comment by douglas125** [comment]
This PR works for me and solved my problem. It would be nice to have it merged.

File: README.md [hunk 1]
```diff
@@ -80,6 +80,13 @@ python main.py test --exp_name laneatt_r34_tusimple
 ```
 The results on TuSimple and LLAMAS should match exactly the ones reported in the paper. The results on CULane will deviate in the order of 0.1% (as shown in the CULane table below), since the metric reported on the paper was computed with the official code (C++), while this script will compute it using our implementation (which is much faster and in Python). The official metric implementation is available [here](https://github.com/XingangPan/SCNN/tree/master/tools/lane_evaluation).
 
+3. Run demo for image or video:
```
Comment (lucastabelini) [inline README.md:83]: Please place those instructions on a separate subsection. The demo is not a step necessary to reproduce the paper's results. Also, keep the commands according to Linux's standards (e.g., paths are divided by a forward slash, CLI options start with -- unless it is the shorter version).

File: demo.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import numpy
+
+import torch
+import cv2
+from lib.config import Config
+import os
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Train lane detector")
```
Comment (lucastabelini) [inline demo.py:11]: Fix the description.

File: demo.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    parser.add_argument(
+        "mode", choices=["video", "image"], help="Video or image?")
+    parser.add_argument("-path", help="image/video path", required=True)
```
Comment (lucastabelini) [inline demo.py:14]: Options should start with `--`.

File: demo.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    parser.add_argument("-cfg", help="Config file", required=True)
+    args = parser.parse_args()
+    return args
+
+
+def process(model, image, img_w, img_h, cfg, device):
+    image = cv2.resize(image, (img_w, img_h))
```
Comment (lucastabelini) [inline demo.py:21]: You should use the `NoLabelDataset` class so that you don't need to repeat the code used to draw predictions nor the steps required to transform raw images to input tensors.

File: demo.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+def main():
+    args = parse_args()
+
+    cfg_path = args.cfg
+    cfg = Config(cfg_path)
+    device = torch.device(
+        'cpu') if not torch.cuda.is_available()else torch.device('cuda')
+
+    model = cfg.get_model()
+    check_point = "./experiments/laneatt_"+"r"+cfg["model"]["parameters"]["backbone"][6:]+"_"+cfg["datasets"]["train"]["parameters"]["dataset"] +"/models/"
```
Comment (lucastabelini) [inline demo.py:55]: Use the `get_checkpoint_path` function in the `Experiment` class for that.

**Review by lucastabelini — CHANGES_REQUESTED** [review]
